contracts/SnappAuction.sol

```diff
@@ -265,4 +268,46 @@ contract SnappAuction is SnappBase {
             appliedAccountStateIndex: 0
         });
     }
+
+    function orderBatchUpdate() private {
+        if (
+            auctionIndex == MAX_UINT ||
+            auctions[auctionIndex].size == maxUnreservedOrderCount() ||
+            block.timestamp > (auctions[auctionIndex].creationTimestamp + 3 minutes)
+        ) {
+            createNewPendingBatch();
+        }
+    }
+
+    function unpackAndVerifyOrderData(bytes memory packedOrder) private view {
+        // TODO - use ECRecover for signatures from first 16 bits of packedOrder
+        uint16 accountId = publicKeyToAccountMap(msg.sender);
+        // uint16 accountId = BytesLib.toUint16(orderData, 0);
+        // WARNING - currently this enables any registered member to place an order for any other registered member!
+
+        uint8 buyToken = BytesLib.toUint8(packedOrder, 2);
+        uint8 sellToken = BytesLib.toUint8(packedOrder, 2 + 1);
+
+        // Would have preferred to use uint 96 or 128, but BytesLib doesn't have this.
+        uint buyAmount = BytesLib.toUint(packedOrder, 2 + 1 + 1);
```

Comment:
I think you can still explicitly cast down to uint96

Reply:
I was not able to cast down to 128, I don't see how I will be able to cast down even more.

Reply:
Does this not work:

```
uint foo = 1;
uint96 bar = uint96(foo);
```

?

Reply:
interesting... I was not aware of this. Thanks.

Reply:
Yes, it turns out, that the bytesLib `toUint` function has a length requirement on the input bytes length as seen here

https://github.com/GNSPS/solidity-bytes-utils/blob/14ca2bd38f96f3b18c06c0cf13df1997a3831a78/contracts/BytesLib.sol#L337-L346

This means I will have to slice and pad the info and then cast down. Luckily, this will allow us to work with the desired 96 bits right off the bat, and no need to fiddle around with 128 checks.